legacy/full/Dockerfile

```
@@ -183,26 +183,26 @@ ENV PATH=/workspace/.rvm/bin:$PATH
 LABEL dazzle/layer=lang-rust
 LABEL dazzle/test=tests/lang-rust.yaml
 USER gitpod
+ENV PATH=$PATH:$HOME/.cargo/bin
 RUN cp /home/gitpod/.profile /home/gitpod/.profile_orig && \
-    curl -fsSL https://sh.rustup.rs | sh -s -- -y --profile minimal --default-toolchain 1.58.0 \
+    curl -fsSL https://sh.rustup.rs | sh -s -- -y --profile minimal --default-toolchain 1.58.1 \
     && .cargo/bin/rustup component add \
         rls \
         rust-analysis \
         rust-src \
         rustfmt \
-    && .cargo/bin/rustup completions bash | sudo tee /etc/bash_completion.d/rustup.bash-completion > /dev/null \
-    && .cargo/bin/rustup completions bash cargo | sudo tee /etc/bash_completion.d/rustup.cargo-bash-completion > /dev/null \
-    && grep -v -F -x -f /home/gitpod/.profile_orig /home/gitpod/.profile > /home/gitpod/.bashrc.d/80-rust
-ENV PATH=$PATH:$HOME/.cargo/bin
-# TODO: setting CARGO_HOME to /workspace/.cargo avoids manual updates. Remove after full workspace backups are GA.
+    && rustup completions bash | sudo tee /etc/bash_completion.d/rustup.bash-completion > /dev/null \
+    && rustup completions bash cargo | sudo tee /etc/bash_completion.d/rustup.cargo-bash-completion > /dev/null \
+    && printf '%s\n' "$(grep -v -F -x -f /home/gitpod/.profile_orig /home/gitpod/.profile)" \
+    'mkdir -m 0755 -p "${CARGO_HOME:-/workspace/.cargo}" 2>/dev/null' > $HOME/.bashrc.d/80-rust \
+    && mkdir -m 0755 -p $HOME/.config/fish \
+    && printf '%s\n' 'alias rustup="env -u CARGO_HOME (command -v rustup)"' >> $HOME/.config/fish/config.fish \
+    && printf '%s\n' 'alias rustup="env -u CARGO_HOME $(command -v rustup)"' >> $HOME/.bashrc.d/80-rust \
+    && printf '%s\n' 'alias rustup="env -u CARGO_HOME $(command -v rustup)"' >> $HOME/.zshrc
+RUN cargo install cargo-watch cargo-edit cargo-tree cargo-workspaces
```

```suggestion
RUN cargo install cargo-watch cargo-edit cargo-workspaces
```

See previous comment.

---

I guess I will no longer update the legacy image.